lib/plugins/nwc/src/persist.rs

```diff
@@ -26,30 +43,180 @@ impl Persister {
+                        min_interval = Some(reset_time_sec);
+                    }
+                }
+            }
+            Ok(min_interval.map(u64::from))
+        };
+        match get_min_connection_interval().await {
+            Ok(Some(mut interval)) => {
+                // We set a minimum of MIN_REFRESH_INTERVAL_SEC to avoid breaking the service by
+                // refreshing too frequently
+                if interval < MIN_REFRESH_INTERVAL_SEC {
+                    interval = MIN_REFRESH_INTERVAL_SEC;
+                }
+                // We add a grace period to avoid races between the interval tick and the connection
+                // expiry/budget refresh
+                Some(interval + REFRESH_INTERVAL_GRACE_PERIOD)
+            }
+            _ => None,
+        }
+    }
+
+    /// Refreshes the active connections (expiry and budget) returning two arrays of the
+    /// corresponding connection names
+    pub(crate) async fn refresh_connections(&self) -> NwcResult<RefreshResult> {
+        let _ = self.lock().await?;
+        let now = utils::now();
+        let mut active_connections = self.list_nwc_connections_inner()?;
+        let mut result = RefreshResult::default();
+        for (name, connection) in active_connections.iter_mut() {
+            // If the connection has expired, mark it for deletion
+            if let Some(expiry) = connection.expiry_time_sec {
+                if now >= connection.created_at + expiry {
+                    result.deleted.push(name.clone());
+                    continue;
+                }
+            }
+            // If the connection's periodic budget has to be updated
+            if let Some(ref mut budget) = connection.periodic_budget {
+                if now >= budget.updated_at + budget.reset_time_sec {
+                    budget.used_budget_sat = 0;
+                    budget.updated_at = now;
+                    result.refreshed.push(name.clone());
+                }
+            }
+        }
+        for name in &result.deleted {
+            active_connections.remove(name);
+        }
+        if !result.refreshed.is_empty() || !result.deleted.is_empty() {
+            self.storage.set_item(
+                KEY_NWC_CONNECTIONS,
+                serde_json::to_string(&active_connections)?,
+            )?;
+        }
+        Ok(result)
+    }
+
+    fn list_nwc_connections_inner(&self) -> NwcResult<HashMap<String, NwcConnection>> {
+        let connections = self
             .storage
-            .get_item(KEY_NWC_URIS)?
+            .get_item(KEY_NWC_CONNECTIONS)?
             .unwrap_or("{}".to_string());
-        let uris = serde_json::from_str(&raw_uris)?;
-        Ok(uris)
+        let connections = serde_json::from_str(&connections)?;
+        Ok(connections)
+    }
+
+    pub(crate) async fn list_nwc_connections(&self) -> NwcResult<HashMap<String, NwcConnection>> {
+        let _ = self.lock().await?;
```

Comment:
Same here, I'm not sure its needed. Only when we plan to add/edit/remove connections